k8s/core/configmap/configmap_lock_v1.go

```diff
@@ -177,6 +171,13 @@ func (c *configMap) refreshLockV1(id string) {
 		select {
 		case <-refresh.C:
 			c.kLockV1.Lock()
+			if c.kLockV1.v1LockGen != lockGen {
+				configMapLog(fn, c.name, id, "", nil).Infof(
+					"Lock generation has changed from %d to %d. Exiting from this refresh lock goroutine...",
+					lockGen, c.kLockV1.v1LockGen)
+				c.kLockV1.Unlock()
```

Comment:
Would we want to painc here? 
Difference would mean re-entrant lock. Should our approach be soft and just log or panic and be assertive about it?

Reply:
This does not mean that the lock was reentered. We don't allow re-entrant lock. That is handled in the Lock() itself. This means that we are the previous refreshLock goroutine and we did not exit yet.